### PR TITLE
#9751: add split_query_key_value_and_split_heads to ttnn and replace …

### PR DESCRIPTION
**Pull request by sjameelTT**
…C++ usages

### Ticket
#9751 

### Problem description
This composite op is only in the ttlib subfolder right now and needs to be moved to the ttnn side to delete the transformers file, which is full of a bunch of ops.

### What's changed
Copied and pasted over the implementation and replaced the C++ usages. A lot of Python usages to go through as well though. 

### Checklist
- [ ] Post commit CI passes
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by boris-drazic — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp [hunk 1]
```diff
@@ -70,4 +70,112 @@ operation::ProgramWithCallbacks ConcatenateHeadsDeviceOperation::create_program(
...
+    TT_FATAL((input_tensor.get_legacy_shape() == tt::tt_metal::Shape({batch_size, 1, 384, 3072})), "Unsupported input shape");
+    TT_FATAL(input_tensor.storage_type() == StorageType::DEVICE, "Operands to TM need to be on device!");
+    TT_FATAL(input_tensor.buffer() != nullptr, "Operands to TM need to be allocated in buffers on device!");
+    TT_FATAL(
+        input_tensor.get_dtype() == tt::tt_metal::DataType::BFLOAT16 ||
+            input_tensor.get_dtype() == tt::tt_metal::DataType::BFLOAT8_B,
+        "Unsupported data format");
+
+    if (input_tensor.is_sharded() == false) {
+        TT_FATAL(batch_size >= 7 && batch_size <= 9, "Input batch size must be between 2 to 9 for bert large TM ops!");
```
Comment (dmakoviichuk-tt) [inline ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp:88]: check is different from the comment.

Comment (sjameelTT) [reply ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp:88]: Updated comment.

**Comment by sjameelTT** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/10100436042

**Review by eyonland — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp
Comment (TT-BrianLiu) [inline ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp:1]: Add usage for `optional_output_tensor` in `validate` and `create_output_tensors`. Take a look at: `ConcatenateHeadsDeviceOperation`.

**Review by TT-BrianLiu — APPROVED** [review]
Looks good. Just address the one comment before merging.

File: ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp
Comment (TT-BrianLiu) [reply ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp:1]: Also, please run the models pipelines as well

Comment (sjameelTT) [reply ttnn/cpp/ttnn/operations/experimental/transformer/device/transformer_device_operation.cpp:1]: Done.

Running pipelines and post commit.

**Pull request merged**